src/otx/core/model/entity/instance_segmentation.py

```diff
@@ -127,22 +134,101 @@ def _export_parameters(self) -> dict[str, Any]:
+        self,
+        inputs: T_OTXBatchDataEntity,
+    ) -> T_OTXBatchPredEntity | OTXBatchLossEntity:
+        """Model forward function."""
+        self.model.explain_fn = self.get_explain_fn()
+
+        # If customize_inputs is overridden
+        if self._customize_inputs != ExplainableOTXInstanceSegModel._customize_inputs:
+            customized_inputs = self._customize_inputs(inputs)
+        else:
+            customized_inputs = {inputs}
+        outputs = self._forward_explain_inst_seg(self.model, **customized_inputs)
+
+        return (
+            self._customize_outputs(outputs, inputs)
+            if self._customize_outputs != ExplainableOTXInstanceSegModel._customize_outputs
+            else outputs
+        )
+
+    @staticmethod
+    def _forward_explain_inst_seg(self, inputs: torch.Tensor, data_samples: SampleList, mode: str) -> dict:
+        """Forward func of the BaseDetector instance, which located in is in ExplainableOTXInstanceSegModel().model."""
+        # Workaround to remove grads for model parameters, since after class patching
+        # convolutions are failing since thay can't process gradients
+        for param in self.parameters():
+            param.requires_grad = False
+
+        x = self.extract_feat(inputs)
+        rpn_results_list = self.rpn_head.predict(x, data_samples, rescale=False)
+        results_list = self.roi_head.predict(x, rpn_results_list, data_samples, rescale=True)
+
+        if isinstance(results_list, tuple) and isinstance(results_list[0], torch.Tensor):  # rewrite
+            # Export case, consists of tensors
+            predictions = results_list
+            # For OV task saliency map are generated on MAPI side
+            saliency_map = torch.empty(1, dtype=torch.uint8)
+            feature_vector = [torch.nn.functional.adaptive_avg_pool2d(f, (1, 1)) for f in x]
+            feature_vector = torch.cat(feature_vector, 1)
+
+        elif isinstance(results_list, list) and isinstance(results_list[0], InstanceData):  # rewrite
+            # Predict case, consists of InstanceData
+            predictions = self.add_pred_to_datasample(data_samples, results_list)
+
+            features_for_sal_map = [data_sample.pred_instances for data_sample in data_samples]
+            saliency_map = self.explain_fn(features_for_sal_map)
+            feature_vector = torch.empty(1, dtype=torch.uint8)
+
+        return {
+            "predictions": predictions,
+            "saliency_map": saliency_map,
+            "feature_vector": feature_vector,
+        }
+
+    def get_explain_fn(self) -> Callable:
+        """Returns explain function."""
         from otx.algo.hooks.recording_forward_hook import MaskRCNNRecordingForwardHook
 
-        self.explain_hook = MaskRCNNRecordingForwardHook.create_and_register_hook(
+        explainer = MaskRCNNRecordingForwardHook.create_and_register_hook(
             num_classes=self.num_classes,
         )
+        return explainer.func
```

Comment:
It is possible to align it with detection: remove create_and_register_hook and create instance right here?

Reply:
Totally missed it, thank you for noticing